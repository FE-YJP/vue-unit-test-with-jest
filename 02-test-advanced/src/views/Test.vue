<template>

  <div>
    <table-test class="mb-4"></table-test>
    <v-divider></v-divider>
    <count-test class="mt-4"></count-test>
  </div>

</template>

<script lang="ts">
import Vue from 'vue'
import tableTest from '@/components/Table/Table.vue'
import countTest from '@/components/Count/Count.vue'

export default Vue.extend({
  name: 'Test',
  components: {
    tableTest,
    countTest
  }
})
</script>
