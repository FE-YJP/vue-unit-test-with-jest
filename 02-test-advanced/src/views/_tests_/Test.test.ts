/**
 * slot测试详解：
 * 本例在mount与shallowMount的区别篇也会讲到。主要是stubs的运用
 *
 * 当组件的slot内容在父组件内定义的时候，如果只用shallowMount的话，snapshot测试并不会包含自定义内容
 * 1. 如果只有此一个组件的话，可以直接使用mount
 * 2. 如果包含多个组件，但是存在自定义内容的只有一个组件，应该用shallowMount + stubs实现--> 本例
 */

import { shallowMount } from '@vue/test-utils'
import Test from '../Test.vue'
import slotTest from '@/components/SlotTest/SlotTest.vue'

describe('views.Test', () => {
  const options = {
    stubs: { slotTest }
  }

  describe('snapshot', () => {
    it('snapshot测试', () => {
      const wrapper = shallowMount(Test, options)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
