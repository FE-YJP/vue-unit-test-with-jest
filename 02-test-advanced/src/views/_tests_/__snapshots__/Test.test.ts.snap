// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views.Test snapshot snapshot测试 1`] = `
<div>
  <table-test-stub
    class="mb-4"
  />
   
  <v-divider-stub />
   
  <count-test-stub
    class="mt-4"
  />
   
  <v-divider-stub />
   
  <picture-test-stub
    class="my-4"
  />
   
  <v-divider-stub />
   
  <div
    class="ma-4"
  >
    
      slot测试自定义内容
    
  </div>
</div>
`;
